src/System.Management.Automation/engine/remoting/commands/PSRemotingCmdlet.cs

```diff
@@ -445,6 +445,40 @@ internal enum VMState
             FastSavingCritical,
         }
 
+#nullable enable
+        /// <summary>
+        /// Get the State property from Get-VM result.
+        /// </summary>
+        /// <param name="value">The raw PSObject as returned by Get-VM.</param>
+        /// <returns>The VMState value of the State property if present and parsable, otherwise null.</returns>
+        internal VMState? GetVMStateProperty(PSObject value)
+        {
+            ReadOnlyPSMemberInfoCollection<PSPropertyInfo> stateProps = value.Properties.Match("State");
+            if (stateProps.Count == 0)
+            {
+                return null;
+            }
+
+            object? rawState = stateProps[0].Value;
```

Comment:
It seems we are expecting the "State" property exists and the code doesn't handle the case when it doesn't exist, so can you get the raw state just like before the change? (I assume `value.Properties["State"]` would throw when the property name doesn't exist, but didn't check to be sure)

```suggestion
            object? rawState =value.Properties["State"].Value;
```

Reply:
Sure I can change it back to just getting the value and having it throw. I added this check before as a way to not have it fail if the returned object was not expected but can easily change it back.

Reply:
Updated to just get the state through the indexer so it'll throw that way now.